PackageViewModel/SignPackageViewModel.cs

```diff
@@ -0,0 +1,363 @@
+        {
+            try
+            {
+                using (var store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+                {
+                    store.Open(OpenFlags.ReadOnly | OpenFlags.OpenExistingOnly);
+
+                    // https://github.com/NuGet/NuGet.Client/blob/adfe6d5c37834e8eb11453518e4508a534c15f8d/src/NuGet.Core/NuGet.Commands/SignCommand/SignCommandRunner.cs#L271-L283
+                    var collection = new X509Certificate2Collection();
+
+                    foreach (var certificate in store.Certificates)
+                    {
+                        if (CertificateUtility.IsValidForPurposeFast(certificate, Oids.CodeSigningEku))
+                        {
+                            collection.Add(certificate);
+                        }
+                    }
+
+                    var certificates = X509Certificate2UI.SelectFromCollection(
+                        collection,
+                        "Choose a Certificate for Package Signing",
+                        "Provide the code signing certificate for signing the package.",
+                        X509SelectionFlag.SingleSelection);
+
+                    if (certificates.Count > 0)
+                    {
+                        Certificate = certificates[0];
+                        CertificateFileName = null;
+                        ShowPassword = false;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                OnError(ex);
+            }
+        }
+
+        private void ShowCertificateCommandExecute()
+        {
+            var certificate = Certificate;
+            if (certificate != null)
+            {
+                try
+                {
+                    X509Certificate2UI.DisplayCertificate(certificate);
+                }
+                catch { }
+            }
+        }
+
+        // https://github.com/NuGet/NuGet.Client/blob/4c0c9658445573845ddbeff5656e4b3129f727a2/src/NuGet.Core/NuGet.Commands/SignCommand/SignCommandRunner.cs
+        public async Task<string> SignPackage()
+        {
+            ShowProgress = true;
+            CanSign = false;
+            Status = "Signing package...";
+
+            try
+            {
+                // change to AuthorSignPackageRequest when NuGet.Client is updated
+                using (var tempCertificate = new X509Certificate2(Certificate))
+                using (var signRequest = new SignPackageRequest(tempCertificate, GetHashAlgorithmName(tempCertificate)))
+                {
+                    SigningUtility.Verify(signRequest, NullLogger.Instance);
+
+                    var packagePath = _packageViewModel.GetCurrentPackageTempFile();
+                    var originalPackageCopyPath = Path.GetTempFileName();
+
+                    File.Copy(packagePath, originalPackageCopyPath, overwrite: true);
+
+                    using (var packageReadStream = File.OpenRead(packagePath))
```

Comment:
The code in here should support overwriting signatures; there's a slightly different code path for that if a package is already signed.

Reply:
Yes, I haven't tested it yet.